### PR TITLE
docs: Docs for local development using build scripts

### PR DESCRIPTION
**Pull request by Bazif-Khan**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Related Issues

fixes: #178

### CODE REVIEW

File: content/docs/sidebar.yaml [hunk 1]
```diff
@@ -27,6 +27,8 @@
       path: /docs/stacks/modify
     - title: Stack Structure
       path: /docs/stacks/stack-structure
+    - title: Running stack locally
```
Comment (skoh7645) [inline content/docs/sidebar.yaml]: nit: Running stack locally -> Running Stacks Locally

File: content/docs/stacks/modify.md [hunk 2]
```diff
@@ -19,34 +19,11 @@ cd stacks
...
-3. Edit `.appsody-config.yaml` to use your built stack:
-```
-stack: <org-name>/<stack-id>
-```
-
-4. Run the template with the modified stack image:
-```
-appsody run 
-``` 
+If you have modified files within `/image` you will need to [build and test](/docs/stacks/run-locally) to make sure the stack and its templates are working as intended.
```
Comment (skoh7645) [inline content/docs/stacks/modify.md]: nit: its -> its'

File: content/docs/stacks/run-locally.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+title: Running stack locally
```
Comment (skoh7645) [inline content/docs/stacks/run-locally.md]: nit: Running stack locally -> Running Stacks Locally

File: content/docs/stacks/run-locally.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+title: Running Stacks Locally
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: I'm not sure on the doc name here as this doc is more about building and testing and not running the stacks.

Possibly "Building and Testing Stacks" ?

File: content/docs/stacks/run-locally.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+path: /docs/stacks/run-locally
+---
+
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: A h1 heading and an introduction would be good here. 

Maybe include an overview of the different options available to build. The background on why you should build and test before contributing a stack might be good to include here also.

File: content/docs/stacks/run-locally.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+## Building a stack locally using build scripts
+
+Once you have updated/created a stack, using a build.sh script wrapper you can build the specific stack images that you want to create/modify. It identifies which stacks need to be built, builds stack images and templates, generates all repo indexes and then tests the stacks and indexes locally.
+
+
+1. Specify the stacks which need to build by setting Environment variable (STACKS_LIST)
+```
+  export STACKS_LIST="incubator/nodejs-loopback"
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: We should use `incubator/nodejs-express`

File: content/docs/stacks/run-locally.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+title: Running Stacks Locally
+path: /docs/stacks/run-locally
+---
+
+## Building a stack locally using build scripts
+
+Once you have updated/created a stack, using a build.sh script wrapper you can build the specific stack images that you want to create/modify. It identifies which stacks need to be built, builds stack images and templates, generates all repo indexes and then tests the stacks and indexes locally.
+
+
+1. Specify the stacks which need to build by setting Environment variable (STACKS_LIST)
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: Specify the stacks which need to build by setting `STACKS_LIST` environment variable.

File: content/docs/stacks/run-locally.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+2. Run build.sh file
+```
+  ./ci/build.sh .
+```
+
+3. Add appsody repository by running:
+
+```
+
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: nit: whitespace

File: content/docs/stacks/run-locally.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  export STACKS_LIST="incubator/nodejs-loopback"
+```
+
+
+By default, all the stacks images will build.
+
+
+2. Run build.sh file
+```
+  ./ci/build.sh .
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: For consistency with other code blocks, please remove tab before the command

File: content/docs/stacks/run-locally.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+3. Add appsody repository by running:
+
+```
+
+  appsody repo add <local-repo-name> file://$PWD/ci/assets/<local-stack-repo>
+```
+Note: `<local-stack-repo>` can be `incubator-local.yaml` or `experimental-local.yaml` or `stable-local.yaml`
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: `incubator-index-local.yaml` or `experimental-index-local.yaml` or `stable-index-local.yaml`

File: content/docs/stacks/run-locally.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+
+For example:
+```
+  appsody repo add my-repo file:///Users/foo/stacks/ci/assets/experimental-index-local.yaml
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: We could use `$pwd` here also so the user can just copy and paste

File: content/docs/stacks/run-locally.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+Once you have updated/created a stack, using a build.sh script wrapper you can build the specific stack images that you want to create/modify. It identifies which stacks need to be built, builds stack images and templates, generates all repo indexes and then tests the stacks and indexes locally.
+
+
+1. Specify the stacks which need to build by setting Environment variable (STACKS_LIST)
+```
+  export STACKS_LIST="incubator/nodejs-loopback"
+```
+
+
+By default, all the stacks images will build.
```
Comment (tomleah) [inline content/docs/stacks/run-locally.md]: The templates are also built by `./build.sh` so we should generalise this to "all the stacks will be built" or specify that the templates are built aswell as the images

**Comment by tomleah** [comment]
I think a lot of the content is here, just needs moving about in some cases

File: content/docs/stacks/run-locally.md (hunk 6 above)
Comment (Bazif-Khan) [reply content/docs/stacks/run-locally.md]: changed

File: content/docs/stacks/run-locally.md (hunk 7 above)
Comment (Bazif-Khan) [reply content/docs/stacks/run-locally.md]: changed

File: content/docs/stacks/run-locally.md (hunk 9 above)
Comment (Bazif-Khan) [reply content/docs/stacks/run-locally.md]: removed

File: content/docs/stacks/run-locally.md (hunk 10 above)
Comment (Bazif-Khan) [reply content/docs/stacks/run-locally.md]: changed

File: content/docs/stacks/run-locally.md (hunk 8 above)
Comment (Bazif-Khan) [reply content/docs/stacks/run-locally.md]: removed

File: content/docs/stacks/run-locally.md (hunk 11 above)
Comment (Bazif-Khan) [reply content/docs/stacks/run-locally.md]: changed

File: content/docs/stacks/build-and-test.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+
+    appsody repo add <local-repo-name> file://$PWD/ci/assets/<local-stack-repo>
+    ```
+
+Note: `<local-stack-repo>` can be `incubator-index-local.yaml` or `experimental-index-local.yaml` or `stable-index-local.yaml`
+
+
+For example:
+    ```
+    appsody repo add my-repo file://$PWD/ci/assets/experimental-index-local.yaml
```
Comment (tomleah) [inline content/docs/stacks/build-and-test.md]: Use `incubator-index-local.yaml` as that is the one that will have content in if following the above steps

File: content/docs/stacks/build-and-test.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+path: /docs/stacks/build-and-test
+---
+
+# Building and Testing Stacks
+
+Once you have updated/created a stack, using a build.sh script wrapper you can build the specific stack images that you want to create/modify. It identifies which stacks need to be built, builds stack images and templates, generates all repo indexes and then tests the stacks and indexes locally.
+
+## Building a stack locally using build scripts
+
+1. Specify the stacks which need to build by setting STACKS_LIST environment variable.
```
Comment (tomleah) [inline content/docs/stacks/build-and-test.md]: STACKS_LIST -> \`STACKS_LIST\`?

File: content/docs/stacks/build-and-test.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+1. Build stack image:
+    ```
+    docker build -t <org-name>/<stack-id> -f Dockerfile-stack .
+    ```
+    You can now have access to the stack image to use and test locally.
+
+### Pointing a template to use a specific stack image
+1. Navigate to template directory
+    ```
+    cd /templates/<template-name>
```
Comment (tomleah) [inline content/docs/stacks/build-and-test.md]: We should probably switch these to relative paths as `/templates` isn't at the root etc

**Pull request merged**